Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/CollectionViews/CollectionView.cs

```diff
@@ -0,0 +1,1281 @@
+        {
+            ISupportIncrementalLoading sourceAsSupportIncrementalLoading = _sourceCollection as ISupportIncrementalLoading;
+            return sourceAsSupportIncrementalLoading?.LoadMoreItemsAsync(count);
+        }
+
+        /// <summary>
+        /// Move <seealso cref="CurrentItem"/> to the first item.
+        /// </summary>
+        /// <returns>True if <seealso cref="CurrentItem"/> points to an item within the view.</returns>
+        public virtual bool MoveCurrentToFirst()
+        {
+            VerifyRefreshNotDeferred();
+
+            return MoveCurrentToPosition(0);
+        }
+
+        /// <summary>
+        /// Move <seealso cref="CurrentItem"/> to the last item.
+        /// </summary>
+        /// <returns>True if <seealso cref="CurrentItem"/> points to an item within the view.</returns>
+        public virtual bool MoveCurrentToLast()
+        {
+            VerifyRefreshNotDeferred();
+
+            return MoveCurrentToPosition(Count - 1);
+        }
+
+        /// <summary>
+        /// Move <seealso cref="CurrentItem"/> to the next item.
+        /// </summary>
+        /// <returns>True if <seealso cref="CurrentItem"/> points to an item within the view.</returns>
+        public virtual bool MoveCurrentToNext()
+        {
+            VerifyRefreshNotDeferred();
+
+            if (CurrentPosition < Count)
+            {
+                return MoveCurrentToPosition(CurrentPosition + 1);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Move <seealso cref="CurrentItem"/> to the previous item.
+        /// </summary>
+        /// <returns>True if <seealso cref="CurrentItem"/> points to an item within the view.</returns>
+        public virtual bool MoveCurrentToPrevious()
+        {
+            VerifyRefreshNotDeferred();
+
+            if (CurrentPosition >= 0)
+            {
+                return MoveCurrentToPosition(CurrentPosition - 1);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Move <seealso cref="CurrentItem"/> to the given item.
+        /// If the item is not found, move to BeforeFirst.
+        /// </summary>
+        /// <param name="item">Move CurrentItem to this item.</param>
+        /// <returns>True if <seealso cref="CurrentItem"/> points to an item within the view.</returns>
+        public virtual bool MoveCurrentTo(object item)
+        {
+            VerifyRefreshNotDeferred();
+
+            // if already on item, don't do anything
+            if (object.Equals(CurrentItem, item))
+            {
+                // also check that we're not fooled by a false null _currentItem
+                if (item != null || IsCurrentInView)
+                {
+                    return IsCurrentInView;
+                }
+            }
+
+            int index = -1;
+#if FEATURE_IEDITABLECOLLECTIONVIEW
```

Comment:
Would it be easier to add all of this Conditional WPF code to a Class for a layer of Abstraction, then it would be easier to port to UWP, or change when/if UWP supports it?

Say `CollectionViewManager`, then you have `CollectionViewManager.SupportsEditable()`, `ColllectionViewManager.SupportsFiltering()`, then `CollectionViewManager.IsAddingNew`, `CollectionViewManager.PassesFilter(item)` etc.

Reply:
That sounds like more code churn than I am willing to take on so late in the game. I don't want to rip out all that commented out code either because one day we may be super pleased it's there when the required interfaces become supported.